llvm/lib/CodeGen/SelectionDAG/InstrEmitter.cpp

```diff
@@ -285,6 +285,27 @@ Register InstrEmitter::getVR(SDValue Op,
   return I->second;
 }
 
+static bool isConvergenceCtrlIntrinsic(SDValue Op) {
+  if (Op->isMachineOpcode()) {
```

Comment:
The not-machine opcode case should never reach InstrEmitter

Reply:
The emitter encounters ISD::CopyFromReg here. I am not so sure about making a special case for it. At least this way, the body of the function matches its name!

Reply:
@arsenm the review already has your approval, but I don't want to submit without closing all comments, especially this discussion about handling non-machine opcodes in instr-emitter.

Reply:
CopyFromReg is fine, but you shouldn't need to handle the pre-selection convergence opcodes